jax/api.py

```diff
@@ -443,6 +466,31 @@ def fun(*args):
   return apply_jaxtree_fun(fun, io_tree, *py_args)
 
 def vjp(fun, *primals):
+  """Compute a (reverse-mode) vector-Jacobian product of `fun`.
+
+  This is a more general form of `grad` that can be used for functions with
+  non-scalar outputs. For most common use cases, you most likely want `grad`
```

Comment:
Maybe delete this paragraph. It's true that `vjp` can handle non-scalar outputs, but so can `jacobian`. And this paragraph leaves off that you might want to use this to control the contangent vector that is pulled back.

I don't think any users will mistakenly use this function over `grad`.

An alternative way to convey this information might be to write something like:

`grad` is implemented as a special case of `vjp`, rougly as:
```python
def grad(f):
  def gradfun(*args):
    _, vjpfun = vjp(f, *args)
    return vjpfun(1.)
  return gradfun
```

Though I'm not sure that's necessary either.